Source/cmake/OptionsGTK.cmake

```diff
@@ -5,6 +5,11 @@ WEBKIT_OPTION_BEGIN()
 
 SET_PROJECT_VERSION(2 37 0)
 
+# This is required because we use the DEPFILE argument to add_custom_command().
+# Remove after upgrading cmake_minimum_required() to 3.20.
+if (${CMAKE_VERSION} VERSION_LESS "3.20" AND NOT ${CMAKE_GENERATOR} STREQUAL "Ninja")
+    message(FATAL_ERROR "Building with Makefiles requires CMake 3.20 or newer. Either enable Ninja by passing -GNinja, or upgrade CMake.")
+endif ()
```

Comment:
Needing CMake 3.20 here _or_ Ninja is somewhat unfortunate; I know a few instances in which the Make generator is used, and CMake is not close to be 3.20. Probably this is fine, though, given that this will not be in a stable release until September 2022 🤔 

Another option would be to conditionally use `DEPFILE` (if the generator is Ninja or CMake 3.20+) because not using it is not a big deal for one-off builds, as typically done from release tarballs. Anyway, I say let's land this as-is for now, and we can implement my suggestion as a follow-up later if ever needed.

Reply:
Also: quite silly that CMake does not ignore `DEPFILE` when the current generator does not support it 🙃

Reply:
I can change it to conditionally use `DEPFILE` if you want, but it's going to be messier.

Since GLib and GTK have both required ninja (via meson) for a long time now, I think this is reasonable. At least it shouldn't cause any trouble for Linux distros or similar projects like yocto or buildstream, as they've had plenty of time to figure out how to handle meson.

Reply:
No need, like I wrote, let's land this as-is. We'll pick up the pieces later if there is any fallout, but I don't expect trouble.